utils/config.ts

```diff
@@ -19,15 +20,26 @@ export const NHPartitionFeatureFlag = t.keyof({
   none: null
 });
 
+export type NotificationHubPartitions = t.TypeOf<
+  typeof notificationHubPartitions
+>;
+const notificationHubPartitions = t.interface({
+  name: NonEmptyString,
+  namespace: NonEmptyString,
+  partitionRegex: NonEmptyString,
+  sharedAccessKey: NonEmptyString
+});
+
+const nhPartitionsReadonlyArray = jsonFromString.pipe(
+  t.readonlyArray(notificationHubPartitions)
+);
+
```

Comment:
Added a new type for reading an array of `notificationHubPartitions` elements.

To reduce character in .env, we  don't take the whole connection string, just the variable parts of it (`namespace` and `sharedAccessKey`)

Not sure about this, but I prefer to have an object containing all related values than other implementations.